modules/nf-core/csvtk/join/main.nf

```diff
@@ -0,0 +1,55 @@
+process CSVTK_JOIN {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::csvtk=0.26.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/csvtk:0.26.0--h9ee0642_0':
+        'biocontainers/csvtk:0.26.0--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(csv)
+    val in_format
+    val out_format
+
+    output:
+    tuple val(meta), path("${prefix}.${out_extension}"), emit: csv
+    path "versions.yml"                                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def delimiter = in_format == "tsv" ? "\t" : (in_format == "csv" ? "," : in_format)
```

Comment:
```suggestion
    def delimiter = csv.extension == "tsv" ? "\t" : ","
```
This gets the extension of the input file and defines the delimiter

Reply:
@nvnieuwk i didn't have a reason behind the logic initially, i was just following the convention of other csvtk modules for consistency. but now that i'm reviewing it, doesn't it make sense to have users configure the delimiter in ext.args? people often use alternative delimiters, or use the `txt` extension when their file is tab-delimited.

Reply:
in some ways i like the old format because it makes the in and out delimiters more deliberate coming from the user, and the module doesn't have to guess which delimiter will be used based on string matching.